File: src/main/java/ro/ubb/dp1819/Catalin Podariu/lab01/src/main/java/model/coffee/CoffeeBuilder.java
package model.coffee;

import model.ingredient.Ingredient;

public class CoffeeBuilder {
	private Coffee coffee;

	public CoffeeBuilder() {
		this.coffee = new Coffee();
	}

	public CoffeeBuilder ingredient(Ingredient ingredient) {
		this.coffee.addIngredient(ingredient);
		return this;
	}
}

File: anaspoiala-932-dotnet/Lab01_CreationalPatterns/Builder/CoffeeBuilder.cs
using CoffeeMachine.Entities;

namespace CoffeeMachine.Builder
{
	public class CoffeeBuilder : ICoffeeBuilder
	{
		private CustomCoffee coffee;

		public CoffeeBuilder()
		{
			coffee = new CustomCoffee();
		}

		public ICoffeeBuilder AddIngredient(CoffeeIngredient ingredient)
		{
			coffee.AddIngredient(ingredient);
			return this;
		}

		public Coffee BuildCoffee()
		{
			return coffee;
		}
	}
}
